src/biogeochem/DUSTMod.F90

```diff
@@ -356,86 +565,222 @@ subroutine DustEmission (bounds, &
          l = patch%landunit(p)
          g = patch%gridcell(p)
 
-         ! only perform the following calculations if lnd_frc_mbl is non-zero 
-
-         if (lnd_frc_mbl(p) > 0.0_r8) then
-
-            ! the following comes from subr. frc_thr_rgh_fct_get
-            ! purpose: compute factor by which surface roughness increases threshold
-            !          friction velocity (currently a constant)
-
-            frc_thr_rgh_fct = 1.0_r8
+         !--------------------------------------------------------------------------------------------------
+         ! put dust emission calculation here to output threshold friction velocity for the whole globe,
+         ! not just when lnd_frc_mbl = 0. Danny M. Leung 27 Nov 2021
+         bd = (1._r8-watsat(c,1))*2.7e3_r8      ![kg m-3] Bulk density of dry surface soil
+         gwc_sfc = h2osoi_vol(c,1)*SHR_CONST_RHOFW/bd    ![kg kg-1] Gravimetric H2O cont
+         if (gwc_sfc > gwc_thr(c)) then
+            frc_thr_wet_fct = sqrt(1.0_r8 + 1.21_r8 * (100.0_r8*(gwc_sfc - gwc_thr(c)))**0.68_r8)
+         else
+            frc_thr_wet_fct = 1.0_r8
+         end if
 
-            ! the following comes from subr. frc_thr_wet_fct_get
-            ! purpose: compute factor by which soil moisture increases threshold friction velocity
-            ! adjust threshold velocity for inhibition by moisture
-            ! modified 4/5/2002 (slevis) to use gravimetric instead of volumetric
-            ! water content
+         ! output moisture variables
+         gwc(p) = gwc_sfc     ! output surface gravimetric water content
+
+         ! slevis: adding liqfrac here, because related to effects from soil water
+         liqfrac = max( 0.0_r8, min( 1.0_r8, h2osoi_liq(c,1) / (h2osoi_ice(c,1)+h2osoi_liq(c,1)+1.0e-6_r8) ) )
+         ! output liquid fraction
+         liq_frac(p) = liqfrac
+
+         !#######################################################################################################
+         ! calculate Shao & Lu (2000) dust emission threshold scheme here
+         ! use tmp1 from DUSTini for Iversen and White I&W (1982) (~75 um is optimal); use tmp2 for S&L (2000) (~80 um is optimal)
+         !#######################################################################################################
+
+         tmp2 = 1.0_r8*sqrt(0.0123_r8 * (dns_slt*grav*130.0e-6_r8 + 1.65e-4_r8/130.0e-6_r8)) ! calculate S&L (2000) scheme here for threshold; gamma = 1.65e-4 following S&L00, D_p = 127 um ~ 130 um following Leung et al. (2022). As this is a global constant, this line can be put outside the loop to save computational power.
```

Comment:
More magic numbers 130.e-6, 0.0123, 1.65e04,

Reply:
I coded these in. I can define them as parameters.